.github/workflows/python-setup.yml

```diff
@@ -2,25 +2,32 @@
 
 name: Python Setup
```

Comment:
I suggest renaming this to something that indicates the purpose of this workflow e.g. 'Run pre-commit' as well as the name of the .github/workflow/.yml file.